main/src/ca/uwaterloo/flix/language/phase/Instances.scala

```diff
@@ -0,0 +1,127 @@
+/*
```

Comment:
With this in its own class, we will also get a better understanding of how expensive (compilation-time wise) these checks are. Right now we don't have many instances, so presumably it is very fast, but that could (slowly) change over time as we add type classes in many places.